dependency-convergence/src/main/java/com/google/cloud/cloudbomcommitcheck/ArtifactData.java

```diff
@@ -0,0 +1,243 @@
+      MetadataXpp3Reader reader = new MetadataXpp3Reader();
+      Metadata metadata = reader.read(new FileInputStream(metadataFile));
+
+      if (metadata.getVersioning() == null) {
+        return new LatestMetadata("", "");
+      }
+
+      String latestVersion;
+      if (metadata.getVersioning().getLatest() == null) {
+        latestVersion = "";
+      } else {
+        latestVersion = metadata.getVersioning().getLatest();
+      }
+
+      String lastUpdated = metadata.getVersioning().getLastUpdated();
+      if (lastUpdated != null && !lastUpdated.isEmpty()) {
+        Date date = dateFormat.parse(lastUpdated);
+        lastUpdated = outputFormat.format(date);
+      } else {
+        lastUpdated = "";
+      }
+      return new LatestMetadata(latestVersion, lastUpdated);
+    } catch (XmlPullParserException | ParseException | IOException ignored) {
+      return new LatestMetadata("", "");
+    }
+  }
+
+  /**
+   * Grabs the Github URL from the POM's scm section. If the URL is not found, then this checks the
+   * associated parent POM (given that the current POM is not already the parent of our artifact)
+   *
+   * @param pomUrl the Maven URL of the artifact's POM file
+   * @return artifact's Github URL
+   */
+  private static String getScmGithubUrl(String pomUrl) {
+    try {
+      File pomFile = File.createTempFile("pomFile", ".xml");
+      pomFile.deleteOnExit();
+      URL url = new URL(pomUrl);
+      ByteStreams.copy(url.openStream(), new FileOutputStream(pomFile));
+      MavenXpp3Reader read = new MavenXpp3Reader();
+      Model model = read.read(new FileInputStream(pomFile));
+      if (model == null) {
+        return null;
+      }
+      return model.getUrl();
+    } catch (XmlPullParserException | IOException ignored) {
+      return null;
+    }
+  }
+
+  private static String generatePomFileUrl(Artifact artifact) {
+    String groupPath = artifact.getGroupId().replace('.', '/');
+    return basePath + "/" + groupPath
+        + "/" + artifact.getArtifactId()
+        + "/" + artifact.getVersion()
+        + "/" + artifact.getArtifactId() + "-" + artifact.getVersion() + ".pom";
+  }
+
+  private static String generateMetadataUrl(Artifact artifact) {
+    String groupPath = artifact.getGroupId().replace('.', '/');
+    return basePath + "/" + groupPath
+        + "/" + artifact.getArtifactId()
+        + "/maven-metadata.xml";
+  }
+
+  private static class LatestMetadata {
+
+    String latestVersion;
+    String lastUpdated;
+
+    LatestMetadata(String latestVersion, String lastUpdated) {
+      this.latestVersion = latestVersion;
+      this.lastUpdated = lastUpdated;
+    }
+  }
+
+  /**
+   * @return version of google-cloud-shared-dependencies in POM file, empty string if
+   * google-cloud-shared-dependencies was not in the POM file, null if the POM file could not be
+   * found
+   */
+  private static String getSharedDependenciesVersion(String pomUrl) {
```

Comment:
Much of this code seems to share functionality with the dashboard, should we combine the artifacts and share logic?